src/initial_conditions/initial_conditions.jl

```diff
@@ -1331,6 +1331,62 @@ function gcm_initial_conditions(external_forcing_file, cfsite_number)
     end
 end
 
+"""
+    ERA5Driven <: InitialCondition
+The `InitialCondition` from a provided ERA5 forcing file, with data type `DType`.
+"""
+struct ERA5Driven <: InitialCondition
+    external_forcing_file::String
+    cfsite_number::String
+end
+
+function (initial_condition::ERA5Driven)(params)
```

Comment:
These seem similar to the GCMDriven initial condition functions -  you can dispatch over functions that perform the same operations in both setups vs. dispatching just on ERA5.